src/moscot/problems/space/_mapping.py

```diff
@@ -299,24 +319,51 @@ def solve(
         - :attr:`solutions` - the :term:`OT` solutions for each subproblem.
         - :attr:`stage` - set to ``'solved'``.
         """
-        return super().solve(  # type: ignore[return-value]
-            alpha=alpha,
-            epsilon=epsilon,
-            tau_a=tau_a,
-            tau_b=tau_b,
-            rank=rank,
-            scale_cost=scale_cost,
-            batch_size=batch_size,
-            stage=stage,
-            initializer=initializer,
-            initializer_kwargs=initializer_kwargs,
-            jit=jit,
-            min_iterations=min_iterations,
-            max_iterations=max_iterations,
-            threshold=threshold,
-            linear_solver_kwargs=linear_solver_kwargs,
-            device=device,
+        solve_kwargs = {
+            "epsilon": epsilon,
+            "tau_a": tau_a,
+            "tau_b": tau_b,
+            "rank": rank,
+            "scale_cost": scale_cost,
+            "batch_size": batch_size,
+            "stage": stage,
+            "initializer": initializer,
+            "initializer_kwargs": initializer_kwargs,
+            "jit": jit,
+            "min_iterations": min_iterations,
+            "max_iterations": max_iterations,
+            "threshold": threshold,
+            "device": device,
             **kwargs,
+        }
+
+        # convert problem type to linear for alpha=0
+        if alpha == 0.0 and self.problem_kind != "linear":
+            if self.filtered_vars and len(self.filtered_vars) == 0:
+                raise ValueError("Expected `var_names` to be non-empty for a :attr:`alpha`=0.")
+            logger.info("Ignoring quadratic terms for :attr:`alpha=0`.")
+            self._problem_kind = "linear"
+            for _, value in self.problems.items():
+                value._x = None
+                value._y = None
+                value._problem_kind = "linear"
+
```

Comment:
Isn't this done automatically? I am asking because I am not sure. If not I think it's better to change behaviour of CompoundProblem because setting this later seems a bit hacky

Reply:
mmh I don't think it is, but also I haven't tested, but I thought that the alpha value would check whether the geometries where correct and if not raised error. Here, it makes sure that the alpha and geometries are correct. I think it would be useful to understand what is the condition for raising error in the OT problem where this is passed though.

Reply:
I checked and it is handled here https://github.com/theislab/moscot/blob/3fd4544b6aa3a13a91dca4f80a85110a879c66ed/src/moscot/base/problems/problem.py#L371-L376. Even if it wasn't we should've modified it that way because in principle we shouldn't access private attributes from this class.